src/readonly-table/index.ts

```diff
@@ -1,220 +1,221 @@
+  if (exportReadOnlyTable) {
+    return;
+  }
+  class KucReadOnlyTable extends KucBase {
+    @property({ type: String, reflect: true, attribute: "class" }) className =
+      "";
+    @property({ type: String, reflect: true, attribute: "id" }) id = "";
+    @property({ type: String }) label = "";
+    @property({ type: Array }) columns: Column[] = [];
+    @property({ type: Array }) data: object[] = [];
+    @property({ type: Boolean }) pagination = true;
+    @property({ type: Number }) rowsPerPage = 5;
+    @property({
+      type: Boolean,
+      attribute: "hidden",
+      reflect: true,
+      converter: visiblePropConverter,
+    })
+    visible = true;
+    @state()
+    private _pagePosition: number = 1;
+    @state()
+    private _columnOrder: string[] = [];
+
+    constructor(props?: ReadOnlyTableProps) {
+      super();
+      if (!props) {
+        return;
+      }
+      const validProps = validateProps(props);
+      Object.assign(this, validProps);
     }
-    if (!Array.isArray(props.columns) && props.columns !== undefined) {
-      throw new Error("'columns' property is invalid");
+
+    shouldUpdate(changedProperties: PropertyValues): boolean {
+      if (changedProperties.has("columns")) {
+        if (!validateColumns(this.columns)) {
+          throwErrorAfterUpdateComplete(
+            this,
+            ERROR_MESSAGE.COLUMNS.IS_NOT_ARRAY
+          );
+          return false;
+        }
+      }
+
+      if (changedProperties.has("data") && !validateData(this.data)) {
+        throwErrorAfterUpdateComplete(this, ERROR_MESSAGE.DATA.IS_NOT_ARRAY);
+        return false;
+      }
+
+      if (changedProperties.has("rowsPerPage")) {
+        if (!validateRowsPerPage(this.rowsPerPage)) {
+          throwErrorAfterUpdateComplete(
+            this,
+            ERROR_MESSAGE.ROWS_PER_PAGE.INVALID
+          );
+          return false;
+        }
+      }
+
+      return true;
     }
-    if (!Array.isArray(props.data) && props.data !== undefined) {
-      throw new Error("'data' property is invalid");
+
+    willUpdate(changedProperties: PropertyValues): void {
+      if (changedProperties.has("columns")) {
+        this._columnOrder = [];
+        this.columns.map((col) =>
+          this._columnOrder.push(col.field ? col.field : "")
+        );
+      }
+
+      if (changedProperties.has("rowsPerPage")) {
+        this.rowsPerPage = Math.round(this.rowsPerPage);
+      }
     }
-    props.data &&
-      props.data.forEach((data) => {
-        if (!Array.isArray(data)) {
-          throw new Error("'data' property is invalid");
-        }
-      });
 
-    const validProps = validateProps(props);
-    Object.assign(this, validProps);
-  }
+    render() {
+      const currentPageData = this._createDisplayData(
+        this.data,
+        this._pagePosition,
+        this.rowsPerPage
+      );
+      return html`
+        <table class="kuc-readonly-table__table" aria-label="${this.label}">
```

Comment:
I think we don't need aria-label because we are using <caption> instead.
https://bozuman.cybozu.com/k/#/space/6073/thread/43185/3465294/6743551
Suggestion from Poca11y team.